src/util/statusSeverity.ts

```diff
@@ -0,0 +1,37 @@
+import { Protocol } from '../protocol/protocol';
+
+export class StatusSeverity {
+
+    private status: Protocol.Status;
+
+    constructor(status: Protocol.Status) {
+        this.status = status;
+    }
+
+    public isOk(): boolean {
+        return this.status
+          && this.status.severity === 0;
+    }
+
+    public isInfo(): boolean {
+        return this.testSeverity(1, this.status);
+    }
+
+    public isWarning(): boolean {
+        return this.testSeverity(2, this.status);
+    }
+
+    public isError(): boolean {
+        return this.testSeverity(4, this.status);
+    }
+
+    public isCancel(): boolean {
+        return this.testSeverity(8, this.status);
+    }
+
+    private testSeverity(bitmask: number, status: Protocol.Status): boolean {
+        console.log(`severity = ${status.severity}, bitmask = ${bitmask}`);
+        return status
+          && (status.severity & bitmask) === bitmask;
```

Comment:
Is there any reason for the bit AND here? Seems to me simple (severity === arg) is sufficient.

Reply:
if you use 1 severity only when creating it you can use ===. But then you can combine those via bitwise OR as the javadoc in server states:
```
	/**
	 * Status type severity (bit mask, value 1) indicating this status is
	 * informational only.
```
ex. ```status.setSeverity(Status.ERROR | Status.CANCEL)```

So to fullfill the API as it is stated, I'm using `&`

Reply:
Haven't read the specs for this, makes sense now. Makes me wonder why somebody would want to do that, but that doesn't invalidate your code.

Reply:
I'm just thinking if it would look better if the methods were static, so you wouldn't need to instantiate for every status. WDYT @adietish ?

Reply:
@jrichter1 yeah, it was afaik never used in that way in our own code, but since it's promised in API we should make our code be able to deal with it.